python/lsst/ts/eas/cmdwrapper.py

```diff
@@ -0,0 +1,272 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["command_wrapper", "close_command_tasks", "DEFAULT_DORMANT_TIME"]
+
+import asyncio
+import functools
+import logging
+from typing import Any, Callable, Coroutine
+
+from lsst.ts import salobj
+from lsst.ts.utils import current_tai
+
+ExceptionCallback = Callable[[Exception], None]
+
+DEFAULT_DORMANT_TIME = 60.0
+TASK_ATTR_REGISTRY = "_command_wrapper_task_attrs"
+
+
+def _command_label(remote: salobj.Remote, command: salobj.topics.RemoteCommand) -> str:
+    """Return a display label for the remote command.
+
+    Parameters
+    ----------
+    remote : `~salobj.Remote`
+        The SAL remote associated with the command.
+    command : `~salobj.topics.RemoteCommand`
+        The command object used to generate the label.
+
+    Returns
+    -------
+    `str`
+        Human-readable label for logging and diagnostics.
+    """
+    remote_name = remote.salinfo.name
+    index = remote.salinfo.index
+    command_name = command.topic_info.attr_name
+    if index is not None and index > 0:
+        return f"{remote_name}:{index}.{command_name}"
+    return f"{remote_name}.{command_name}"
+
+
+async def _run_command(
+    *,
+    log: logging.Logger,
+    remote: salobj.Remote,
+    command: salobj.topics.RemoteCommand,
+    kwargs_list: list[dict[str, Any]],
+    timeout: float | None,
+    dormant_time: float,
+    exception_callback: ExceptionCallback | None,
+) -> None:
+    """Wait for ENABLED, issue command calls, and handle errors.
+
+    Parameters
+    ----------
+    log : `logging.Logger`
+        Logger to use for warnings and exceptions.
+    remote : `~salobj.Remote`
+        Remote whose summary state gates command issuance.
+    command : `~salobj.topics.RemoteCommand`
+        Command to invoke once the remote is ENABLED.
+    kwargs_list : `list` [`dict` [`str`, `Any`]]
+        Sequence of keyword argument mappings for each command call.
+    timeout : `float` | `None`
+        Maximum time (seconds) to wait for ENABLED, or None to wait
+        indefinitely.
+    dormant_time : `float`
+        Sleep interval between summary state checks.
+    exception_callback : `ExceptionCallback` | `None`
+        Optional callback invoked when the command raises an exception.
+    """
+    label = _command_label(remote, command)
+    deadline = None if timeout is None else current_tai() + timeout
+
+    try:
+        while deadline is None or deadline > current_tai():
```

Comment:
"aget" can be (I hope) used instead of the busy-wait get. That will also remove "deadline":
```
while timeout is None or timeout > 0:
    last = time.time()
    summary_state = await remote.evt_summaryState.aget(timeout)
    if timeout is not None:
        timeout -= time.time() - last
    if summary_state is not None and summary_state.summaryState == salobj.State.ENABLED:
        for kwargs in kwargs_list:
            await command.set_start(**kwargs)
        return
```
There might be some problem I oversaw, but still I bet it's worth to try.